src/score-calculator.ts

```diff
@@ -119,24 +241,152 @@ const roundUp = (input: number): number => {
     : (Math.floor(intInput / 10000) + 1) / 10;
 };
 
+// populate temp and env metrics if not provided
+export const populateUndefinedMetrics = (
+  metricsMap: Map<Metric, MetricValue>
+): Map<Metric, MetricValue> => {
+  [...temporalMetricMap, ...environmentalMetricMap].map((metric) => {
+    if (![...metricsMap.keys()].includes(metric)) {
+      metricsMap.set(
+        metric,
+        metricsIndex[metric] ? metricsMap.get(metricsIndex[metric]) : 'X'
+      );
+    }
+    if (metricsMap.get(metric) === 'X') {
+      metricsMap.set(
+        metric,
+        metricsMap.get(metricsIndex[metric])
+          ? metricsMap.get(metricsIndex[metric])
+          : 'X'
+      );
+    }
+  });
+
+  return metricsMap;
+};
+
+export type ScoreResult = {
+  score: number;
+  impact: number;
+  exploitability: number;
+  metricsMap: Map<Metric, MetricValue>;
+};
```

Comment:
i'm not sure that these details are required from the point of external API: perhaps we should leave only calculateXScore methods exported (with a single number as result). 
Also as we have now temporal and environmental scores, having aggregated `calculateScore` with a triple as result would be nice.